File: 3163-string-compression-iii/3163-string-compression-iii.cpp
class Solution {
public:
    string compressedString(string word) {
        int n = word.size();
        string comp = "";
        int i = 0;
        while(i < n){
            int count = 1;
            char ch = word[i];
            i++;
            while(i < n && count < 9 && word[i] == ch){
                count++;
                i++;
            }
            comp += to_string(count) + ch;
        }
        return comp;
    }
};

File: 3163-string-compression-iii/3163-string-compression-iii.java
class Solution {
    public String compressedString(String word) {
        int n = word.length();
        
        StringBuilder compressedStr = new StringBuilder();
        
        char prevChar = word.charAt(0);
        int prevCharCount = 1;
        
        for(int i = 1; i < n; i++) {
            if(prevCharCount < 9 && prevChar == word.charAt(i)) {
                prevCharCount++;
            } else {
                compressedStr.append((char)(prevCharCount + '0'));
                compressedStr.append(prevChar);
                
                prevChar = word.charAt(i);
                prevCharCount = 1;
            }
        }
        
        compressedStr.append((char)(prevCharCount + '0'));
        compressedStr.append(prevChar);
        
        return compressedStr.toString();
    }
}